### PR TITLE
add: install-deep-semgrep subcommand

### PR DESCRIPTION
**Pull request by brendongo**
Test plan:
run `semgrep install-deep-semgrep` locally (pointed to dev-stacks for now since endpoints not released to prod yet)

PR checklist:

- [x] Documentation is up-to-date
- [x] Changelog is up-to-date
- [x] Change has no security implications (otherwise, ping security team)

### CODE REVIEW

File: semgrep/tests/e2e/snapshots/test_help/test_help_text/--help/help.txt [hunk 1]
```diff
@@ -7,6 +7,7 @@ Options:
 
 Commands:
   ci       The recommended way to run semgrep in CI
+  install  Install specific semgrep dependencies (Experimental)
```
Comment (ievans) [inline semgrep/tests/e2e/snapshots/test_help/test_help_text/--help/help.txt]: Suggestion: remove from help so it's obvious this is experimental and users don't put it in scripts

File: semgrep/semgrep/commands/install.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+from semgrep.error import INVALID_API_KEY_EXIT_CODE
+from semgrep.semgrep_core import SemgrepCore
+from semgrep.util import set_flags
+from semgrep.verbose_logging import getLogger
+
+logger = getLogger(__name__)
+
+
+@click.group()
+def install() -> None:
```
Comment (ievans) [inline semgrep/semgrep/commands/install.py]: rename to `deep-semgrep-install` or something?

Comment (brendongo) [reply semgrep/semgrep/commands/install.py]: Renamed to install-deep-semgrep

File: semgrep/semgrep/commands/install.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        file_size = int(r.headers.get("Content-Length", 0))
+
+        with open(deep_semgrep_path, "wb") as f:
+            with tqdm.wrapattr(r.raw, "read", total=file_size) as r_raw:
+                shutil.copyfileobj(r_raw, f)
+
+    os.chmod(
+        deep_semgrep_path,
+        os.stat(deep_semgrep_path).st_mode | stat.S_IEXEC | stat.S_IXGRP | stat.S_IXOTH,
+    )
```
Comment (underyx) [inline semgrep/semgrep/commands/install.py:97]: Asking just to learn: I'd write this as just `deep_semgrep_path.chmod(0o744)`. What does all the st_mode stuff do? I never saw it before.

**Review by underyx — APPROVED** [review]

File: semgrep/semgrep/commands/install.py (hunk 3 above)
Comment (brendongo) [reply semgrep/semgrep/commands/install.py:97]: It's the bit wise stuff. so if you print it out they're probably 0b00010000 or something so if you or them they would become 744 but I did this OR-ing to keep whatever the other permissions were and just modified the execute permissions

**Comment by brendongo** [comment]
This is waiting on `deep-semgrep --version` functionaity

**Comment by brendongo** [comment]
--version functionality is in

**Pull request merged**